File: app/src/main/java/com/dev/lishabora/Utils/ApproveFarmerPayCardListener.java
package com.dev.lishabora.Utils;

import com.dev.lishabora.Models.PayoutFarmersCollectionModel;

public interface ApproveFarmerPayCardListener {
    void onApprove(PayoutFarmersCollectionModel model, Double totalKshToPay, Double toLoanInstallmentPayment, Double toOrderInstallmentPayment);

    void onApprovePayLoan(PayoutFarmersCollectionModel model, Double totalKshToPay, Double toLoanInstallmentPayment);

    void onApprovePayOrder(PayoutFarmersCollectionModel model, Double totalKshToPay, Double toOrderInstallmentPayment);

    void onApprove(PayoutFarmersCollectionModel model, Double totalKshToPay);

    void onApprove(PayoutFarmersCollectionModel model);

    void onApproveError(String error);

    void onApproveDismiss();
}

File: app/src/main/java/com/dev/lishabora/Utils/ApproveFarmerPayCardListener.kt
package com.dev.lishabora.Utils

import com.dev.lishabora.Models.PayoutFarmersCollectionModel

interface ApproveFarmerPayCardListener {
    fun onApprove(farmerBalance: Double, model: PayoutFarmersCollectionModel, totalKshToPay: Double?, toLoanInstallmentPayment: Double?, toOrderInstallmentPayment: Double?)

    fun onApprovePayLoan(farmerBalance: Double, model: PayoutFarmersCollectionModel, totalKshToPay: Double?, toLoanInstallmentPayment: Double?)

    fun onApprovePayOrder(farmerBalance: Double, model: PayoutFarmersCollectionModel, totalKshToPay: Double?, toOrderInstallmentPayment: Double?)

    fun onApprove(farmerBalance: Double, model: PayoutFarmersCollectionModel, totalKshToPay: Double?)

    fun onApprove(farmerBalance: Double, model: PayoutFarmersCollectionModel)

    fun onApproveError(error: String)

    fun onApproveDismiss()

}
